src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs

```diff
@@ -61,45 +65,97 @@ public override void ConfigureServices(IServiceCollection services)
             services.TryAddEnumerable(new[]
             {
                 // Orchard-specific initializers:
-                ServiceDescriptor.Transient<IConfigureOptions<AuthenticationOptions>, OpenIdClientConfiguration>(),
-                ServiceDescriptor.Transient<IConfigureOptions<OpenIdConnectOptions>, OpenIdClientConfiguration>(),
+                ServiceDescriptor.Singleton<IConfigureOptions<AuthenticationOptions>, OpenIdClientConfiguration>(),
```

Comment:
Options initializers (post or not) are always registered as singletons. Not sure why we opted for transient - it makes no sense - but I fixed it.

Reply:
i don't recall exactly but i had fixed an issue elsewhere for an `IConfigureOptions` by using transient in place of singleton. I think it is the same concerns as other services, thread safety, short lived resources and so on. That said, mvc uses a lot transient for `IConfigureOptions` see here.

https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/MvcCoreServiceCollectionExtensions.cs#L146

Reply:
I guess it's another consistency as the options stack and the security handlers all use singletons: 
  - https://github.com/aspnet/Options/blob/master/src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs#L72
  - https://github.com/aspnet/Security/blob/master/src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerExtensions.cs#L25

Anyway, the practical impact is extremely small as options are always cached anyway, so the initializers are only resolved once: whether they are transient or singletons is not really relevant.

Reply:
Okay cool.